### PR TITLE
Update Node versions in travis CI & Update yeoman stack

### PR DESCRIPTION
**Pull request by janizde**
Updates Node versions in travis configs, updates the used yeoman packages and fixes a bug with `package.json:name`

* Update `node_js` versions in `.travis.yml` of the generator as well as the generated package to `10`, `12`, `14` according to [the current LTS version landscape](https://nodejs.org/en/about/releases/)
* Updates `yeoman` package
  * Add dependencies of `yeoman-assert` and `yeoman-test` (these were extracted in the meantime)
  * Adopt generator code to new version (e.g. use `class extends Generator` instead of `Generator.extend({})`)
  * Remove `filter` option of `keywords` prompt and move to `.then()` callback (`filter` doesn't seem to be supported anymore)
* Fix an issue that would cause `name` not being set in `package.json` causing `npm ci` to fail with a weird error
* Bump version of generator `2.0.1 -> 2.1.0`
* Replace `installDependencies` with `npmInstall` (the prior used to try to run `bower` and fail 🤷 )

If this get's through, I'm happy to follow up with a PR moving to Jest for both generator and generated package as well as latest Babel, since I found this generator to be super helpful!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "generator-babel-plugin",
-  "version": "2.0.1",
```
Comment (JLHwung) [inline package.json:3]: Please leave `version` as-is, we will bump this version when published.

File: package.json [hunk 2]
```diff
@@ -20,10 +20,12 @@
     "yeoman-generator"
   ],
   "dependencies": {
-    "lodash": "^4.7.11",
-    "yeoman-generator": "^0.19.0"
+    "lodash": "^4.17.19",
+    "yeoman-generator": "^4.11.0"
```
Comment (JLHwung) [inline package.json:24]: Does `yeoman-generator` bump its node depedencies? If it does it is a breaking change and we should do a new major release.

File: package.json (hunk 1 above)
Comment (janizde) [reply package.json]: Right, updated!

File: package.json (hunk 2 above)
Comment (janizde) [reply package.json:24]: Yes a lot has changed between these as you can see in the changes that had to be made to the generators, also their dependencies changed, so it's breaking changes on yeoman side, wasn't sure how to reflect this in the generator itself.

Comment (janizde) [reply package.json:24]: If we want to move to latest babel and jest in a follow up PR it might be worth combining the two in one major update?

Comment (JLHwung) [reply package.json:24]: > how to reflect this in the generator itself.

`generator-babel-plugin` works for specific node versions. If by upgrading `yeoman-generator`, some legacy node versions turns unsupported, we should clearly say it and mark it as breaking changes.

> combining the two in one major update?

Sure we can release a new major after the overhauling is done. Thanks!

**Comment by janizde** [comment]
@JLHwung I added the changes for jest and babel to a separate branch https://github.com/janizde/generator-babel-plugin/compare/bump-node-version...janizde:modernize-stack What would be the best way to proceed you think? Merge this one first and then open a new PR?

**Comment by JLHwung** [comment]
@janizde You can rebase on this branch and submit a new PR targets to `master`, and mentions that "this PR includes commits from #XXXX".

**Pull request merged**